packages/cli/src/utils.ts

```diff
@@ -84,37 +89,97 @@ export function exportsField(
+          // or maybe we just generate more .d.ts files in the dist rather than having a types condition
+          types:
+            exportsFieldConfig.importConditionDefaultExport === "default"
+              ? {
+                  import: getExportsFieldOutputPathForConditions(entrypoint, [
+                    "import",
+                  ]),
+                  default: getExportsFieldOutputPathForConditions(
+                    entrypoint,
+                    []
+                  ),
+                }
+              : getExportsFieldOutputPathForConditions(entrypoint, []),
+        }),
+        ...createExportsField(
+          exportsFieldConfig.conditions.groups,
+          (conditions) => ({
+            module: getExportsFieldOutputPathForConditions(
+              entrypoint,
+              conditions.concat("module")
+            ),
+            ...(exportsFieldConfig.importConditionDefaultExport ===
+              "default" && {
+              import: getExportsFieldOutputPathForConditions(
+                entrypoint,
+                conditions.concat("import")
+              ),
+            }),
+            default: getExportsFieldOutputPathForConditions(
+              entrypoint,
+              conditions
+            ),
+          })
+        ),
+      };
+    }
+  }
+
+  return {
+    ...output,
+    "./package.json": "./package.json",
+    ...exportsFieldConfig.extra,
+  };
+}
 
+function exportsFieldForLegacyConditions(
+  pkg: Package,
+  envs: ({
+    kind: "legacy";
+  } & CanonicalExportsFieldConfig["conditions"])["envs"],
+  importConditionDefaultExport: CanonicalExportsFieldConfig["importConditionDefaultExport"]
+) {
   let output: Record<string, ExportsConditions> = {};
-  pkg.entrypoints.forEach((entrypoint) => {
+  for (const entrypoint of pkg.entrypoints) {
     const esmBuild = getExportsFieldOutputPath(entrypoint, "esm");
     const exportConditions = {
-      module: exportsFieldConfig.envConditions.size
+      module: envs.size
         ? {
-            ...(exportsFieldConfig.envConditions.has("worker") && {
+            ...(envs.has("worker") && {
               worker: getExportsFieldOutputPath(entrypoint, "worker"),
             }),
-            ...(exportsFieldConfig.envConditions.has("browser") && {
+            ...(envs.has("browser") && {
               browser: getExportsFieldOutputPath(entrypoint, "browser-esm"),
             }),
             default: esmBuild,
           }
         : esmBuild,
-      ...(exportsFieldConfig.importConditionDefaultExport === "default" && {
+      ...(importConditionDefaultExport === "default" && {
         import: getExportsImportUnwrappingDefaultOutputPath(entrypoint),
       }),
       default: getExportsFieldOutputPath(entrypoint, "cjs"),
     };
 
-    output[
-      "." + entrypoint.name.replace(entrypoint.package.name, "")
-    ] = exportConditions;
-  });
-  return {
-    ...output,
-    "./package.json": "./package.json",
-    ...exportsFieldConfig.extra,
```

Comment:
what has happened to those in the legacy mode? I mean, likely this works OK since tests pass but i wonder implementation-wise